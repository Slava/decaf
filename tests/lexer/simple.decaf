/* keywords and operators */
int double string char bool null class interface this extends implements for while if else return break new , () {}[];=.&&||!+-*/% >< >= <= ==

/* ids */
identifier another one

/* some comments go here, should be ignored */

/* literals */
"some string"
123 32495392 123 012323
222.45 0.1  1. 0.324

