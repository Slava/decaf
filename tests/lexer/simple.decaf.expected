type int
type double
type string
id char
type bool
null
class
interface
this
extends
implements
for
while
if
else
return
break
new
comma
paren_open
paren_close
brace_open
brace_close
array_decl
semicolon
equals
dot
binary_logical_op(&&)
binary_logical_op(||)
unary_logical_op(!)
binary_unary_add_op(+)
binary_unary_add_op(-)
binary_mult_op(*)
binary_mult_op(/)
binary_mult_op(%)
binary_cmp_op(>)
binary_cmp_op(<)
binary_cmp_op(>=)
binary_cmp_op(<=)
binary_cmp_op(==)
id identifier
id another
id one
str some string
int 123
int 32495392
int 123
int 12323
double 222.45
double 0.1
double 1.
double 0.324
eof
